File: AuraFit-BackEnd/src/main/java/app/aurafitbackend/Beans/FitType.java
package app.aurafitbackend.Beans;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "fit_types")
@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder

/**
 *
 * this uses a table for get a fit for clothing type so if the productvariant is clothing
 * get SLIMFIT REGULARFIT etc..
 * for socks - ankle socks long socks etc...
 */

public class FitType {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    private Long id;

    private String name;

    @ManyToOne
    private ProductType productType;

}

File: aurafit-frontend/src/Models/FitType.ts
import {ProductType} from "./ProductType.ts";

export class FitType {
    id: number
    name: string
    productType: ProductType

    constructor(id:number, name: string, productType: ProductType) {
        this.id = id
        this.name = name
        this.productType = productType
    }
}